assets/RNAseq_report.Rmd

```diff
@@ -514,6 +1290,280 @@ cat("Plots for the [genes of interest](./differential_gene_expression/metadata/g
 
 <!-- In case KEGG analysis was also included; differentiating the cases with and without contrasts -->
 
+
+<!-- PA start -->
+```{r load_PA_libs, eval=params$pathway_analysis, echo=FALSE, message=FALSE, warning=FALSE, results = 'hide'}
+# Load libs invisibly
+invisible(lapply(c(
+"gprofiler2",
+"reshape2",
+"pathview",
+"AnnotationDbi",
+"optparse",
+"org.Mm.eg.db",
+"org.Hs.eg.db"
+), library, character.only=T))
+# Need to load library for your species! Or provide as parameter to biocmanager
+
+organism <- tolower(params$organism)
+short_organism_name <- substr(organism,1,3)
+species_dir <- tempdir()
+if (!require("BiocManager", quietly = TRUE)) {
+    install.packages("BiocManager")
+}
+
+BiocManager::install(params$species_library, lib=species_dir, version="3.14")
+library(params$species_library, lib.loc=species_dir, character.only=T)
+library <- get(params$species_library)
+
+# Default to 1 for the nb of genes DE in a pathway
+min_DEG_pathway <- as.integer(params$min_DEG_pathway)
+# Contrast files
+contrast_files <- list.files('differential_gene_expression/DE_genes_tables/')
+path_contrasts <- 'differential_gene_expression/DE_genes_tables/'
+
+# Reading metadata table
+metadata <- metadata_PA
+metadata$Secondary.Name <- gsub(" ; ", "_", metadata$Secondary.Name) # Remove blank spaces and ; from secondary name
+metadata$Secondary.Name <- gsub(" ", "_", metadata$Secondary.Name) # Remove blank spaces if there from secondary name
+
+# Create output directory
+dir.create("pathway_analysis")
+
+# ------------------
+# Set default params
+# ------------------
+
+# gprofiler pathway / term sources parameters
+datasources <- c("KEGG", "REAC")
+
+# Set theme for graphs
+theme_set(theme_classic())
+
+# ----------------------
+# Start pathway analysis
+# ----------------------
+
+# For each contrast do pathway analysis
+
+# TODO debugging summary table
+# Defining summary variables
+# contrast <- c()
+# number_DE_genes <- c()
+# number_enriched_pathways <- c()
+# list_DE_genes <- list()
+# list_enriched_pathways <- list()
```

Comment:
Can this be removed?

Reply:
Hm, I don't know :'D This part was already commented out when I started working on deseq